hotham/src/systems/stage.rs

```diff
@@ -0,0 +1,40 @@
+use crate::{
+    components::{Hand, Parent, Stage, Transform, TransformMatrix},
+    hecs::{CommandBuffer, Entity, With, World},
+    util::{isometry_to_posef, matrix_to_isometry, posef_to_isometry},
+    xr,
+};
+
+/// Setup Stage entities to track player's frame of reference in the world
+///
+/// This should be run after `add_hand`.
+pub fn add_stage(world: &mut World) -> Entity {
+    // Add Stage entity to track the position of the player's space with respect to the game-world
+    let stage_e = world.spawn((Stage {}, TransformMatrix::default(), Transform::default()));
+
+    // Make hands children of Stage
+    let mut cmd_buffer = CommandBuffer::new();
+    for (hand_e, _) in world.query_mut::<&Hand>() {
+        cmd_buffer.insert(hand_e, (Parent(stage_e),));
+    }
+    cmd_buffer.run_on(world);
+
+    stage_e
+}
+
+/// Update player's views to take into account the current position of the Stage in the world
+///
+/// Must happen each tick after parent transforms have been updated.
+pub fn update_views_with_stage_transform(world: &mut World, views: &mut [xr::View]) {
```

Comment:
This feels super gross (going from `TransformMatrix` to `Isometry3`) but I suppose it's kind of unavoidable until #291 ships. ☹️

Reply:
it is mostly just lazy/simpler to reason about imo. We don't need any of the extra abilities that a matrix gives you, and we ultimately have to cast to a posef anyway, which basically is an isometry

Reply:
I think it would make sense to let `xr:View` keep `stage_from_view` and add something else that keeps track of `world_from_view`.

Reply:
My understanding of how this works is openxr hands us a new view each tick (ideally right before we're about to render), then we either change the views using this system.. or we don't.. Then we just pass them directly to the renderer

I'm happy to have this take a `&xr::View` instead of a `&mut` and return a new owned `xr::View` so you can have a view with respect to both World and Stage, but I'm curious what's the usecase?